### PR TITLE
Baselines the MAINTAINERS.md and CODEOWNERS files.

### LINKED ISSUE #45: Baseline MAINTAINERS, CODEOWNERS and external collaborator permissions
Follow https://github.com/opensearch-project/.github/issues/125 to baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions.

Close this issue when:

1. [MAINTAINERS.md](MAINTAINERS.md) has the correct list of project maintainers.
2. [CODEOWNERS](CODEOWNERS) exists and has the correct list of aliases.
3. Repo permissions only contain individual aliases as collaborators with maintain rights, admin, and triage teams. 
4. All other teams are removed from repo permissions.

If this repo's permissions was already baselined, please confirm the above when closing this issue.

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

This baselines the list of maintainers for the project. It also updates the CODEOWNERS file to use specific names. This is following the pattern supplied in https://github.com/opensearch-project/.github/issues/125 for correcting a repo.

### Issues Resolved

Resolves #45 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has documentation added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/CODEOWNERS
Comment (dblock) [inline .github/CODEOWNERS:1]: GH is complaining, I think you have to remove empty lines.

**Review by dblock — COMMENTED** [review]
LGTM if you want to fix CODEOWNERS. I'll help adjust permissions.

File: .github/CODEOWNERS
Comment (dlvenable) [reply .github/CODEOWNERS:1]: I believe this is because some of these users have not yet been added as repo admins.

> Unknown owner on line N: make sure @USER exists and has write access to the repository

We will need to grant them write permissions to the repository before merging this PR. But, I was waiting on approval.

**Review by sshivanii — APPROVED** [review]

**Review by oeyh — APPROVED** [review]

**Review by naveentatikonda — APPROVED** [review]

**Pull request merged**
